### PR TITLE
Disabled Action menu items by default, CDN available and tooltip on keyboard focus

### PR DESCRIPTION
**Pull request by jongund**
Summary of changes:
* Popup tooltip shows accesskey to open menu when button on hover or focus.
* Popup tooltip is only displayed when a known accesskey is supported by the browser and device operating system.
* Popup tooltip supports high contrast operating system settings.
* Action menu items are disabled by default.
* M of N items in landmark or heading list is disabled by default.
* CDN reference to SkipTo.min.js is now available from University of Illinois.

### CODE REVIEW

**Comment by brianteeman** [comment]
As this breaks backwards compatibility it needs to be a major version ie 5.0

**Comment by jongund** [comment]
What does it break?  It just changes behavior to what most people want.

**Comment by brianteeman** [comment]
@jongund 
> It just changes behavior to what most people want.

Firstly you can never know what most people want. Currently there are over 200 downloads each week. When Joomla 4 is shipped then this will be installed on a massive number of websites and not just the handful of websites that you manage.

Semantic versioning has very specific numbering rules. As you state this is changing the default behaviour then by the very definition that is not backwards compatible. That is you cannot simply upgrade and everything works the same.

https://semver.org/

Given a version number MAJOR.MINOR.PATCH, increment the:

- MAJOR version when you make incompatible API changes,
- MINOR version when you add functionality in a backwards compatible manner, and
- PATCH version when you make backwards compatible bug fixes.

File: src/js/skipto.js [hunk 1]
```diff
@@ -186,26 +201,60 @@
...
+      } else {
+        this.tooltipNode.textContent = tooltip;
+        buttonNode.setAttribute('aria-describedby', id);
+      }
+
+      attachNode.appendChild(this.tooltipNode);
+      this.tooltipNode.style.left = this.tooltipLeft + 'px';
+      this.tooltipNode.style.top = this.tooltipTop + 'px';
+
+      // Need to temporarily show the tooltip to get it's height
```
Comment (brianteeman) [inline src/js/skipto.js:252]:
```suggestion
      // Need to temporarily show the tooltip to get its height

```

https://www.grammarly.com/blog/its-vs-its/

**Comment by jongund** [comment]
@mpnkhan 
I have made the edits to the readme and the CSS, do you have any thoughts on the version numbering?

There are no real changes in functionality, just default configuration changes, that @brianteeman had already requested in issue #75.  Many other people requested this same change to the default configuration, so it seems to me to be just a PATCH to the default configuration.  It is also closer to the configuration of previous versions of SkipTo.

File: README.md [hunk 2]
```diff
@@ -34,7 +36,7 @@ Access keys work  just like regular shortcut keys except that they need a browse
 ## How it works
 
 1. The SkipTo menu button should be the first tabable element on the page, and if it is configured not to be visible when the page is loaded, the menu button becomes visible when it receives focus.
-2. Once the keyboard focus is on the menu button, pressing the ENTER or the SPACEBAR key will pull down the list of important landmarks and headings on the page.
+2. Once the keyboard focus is on the menu button, pressing the ENTER, SPACE, DOWN ARROW or UP ARROW keys will pull down the list of important landmarks and headings on the page.  The button based on the ARIA Authoring Practice [design pattern for menu button](https://w3c.github.io/aria-practices/#menubutton).
```
Comment (brianteeman) [inline README.md:39]:
```suggestion
2. Once the keyboard focus is on the menu button, pressing the ENTER, SPACE, DOWN ARROW or UP ARROW keys will pull down the list of important landmarks and headings on the page.  The button is based on the ARIA Authoring Practice [design pattern for menu button](https://w3c.github.io/aria-practices/#menubutton).
```

File: README.md [hunk 3]
```diff
@@ -400,6 +374,15 @@ Happy skipping!
 
 ## Version History
 
+### Version 4.0.4
+* Popup tooltip shows accesskey to open menu when button on hover or focus.
+* Popup tooltip is only displayed when a known accesskey is supported by the browser and device operating system.
+* Popup tooltip supports high contrast operating system settings.
+* Action menu items are disabled by default.
+* M of N items in landmark orheading list is disabled by default.
```
Comment (brianteeman) [inline README.md:382]:
```suggestion
* M of N items in landmark or heading list is disabled by default.
```

Comment (jongund) [reply README.md:382]: fixed

**Comment by brianteeman** [comment]
OR how about holding of on this PR and making it the first PR in the new repo and call the new skipto in its new home v5

**Comment by jongund** [comment]
Still discussing whether to actually move the production code, I don't see a lot of further development in the near future, so maybe no immediate need to move it.  Need to discuss with @mpnkhan  and some other people.  So no need to hold up this pull request for that.

**Comment by brianteeman** [comment]
With my Joomla hat on the sooner the better or not at all. We hope to release Joomla 4 (the version that includes this) in 4 weeks and changing to a new package source and maybe even package name will be a world of hurt.

**Comment by jongund** [comment]
I think we should know within the next week.

**Comment by mpnkhan** [comment]
@brianteeman I'm unable to npm publish. I have added you. So going forward please take care of publishing.

**Comment by brianteeman** [comment]
And published. Once I remembered to `npm login` it was easy ;)

**Pull request merged**